airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py

```diff
@@ -180,106 +176,34 @@ def test_oauth_flow_parameters(self, actual_connector_spec: ConnectorSpecificati
-            if isinstance(change.t1, list):
-                assert isinstance(
-                    change.t2, str
-                ), f"The current spec change a type field from list to an invalid value:  {spec_diff.pretty()}"
-                assert len(change.t1) == 1 and change.t2 == change.t1[0], f"The current spec narrowed a field type: {spec_diff.pretty()}"
-
-        # Detect when field was made not nullable but is still a list: e.g ["string", "null"] -> ["string"]
-        removed_nullable = [
-            change for change in spec_diff.get("iterable_item_removed", []) if change.path(output_format="list")[-2] == "type"
-        ]
-        assert len(removed_nullable) == 0, f"The current spec narrowed a field type: {spec_diff.pretty()}"
-
-    @pytest.mark.default_timeout(60)
-    @pytest.mark.backward_compatibility
-    def test_enum_field_has_narrowed(self, spec_diff: DeepDiff):
-        """Check if the list of values in a enum was shortened in a spec."""
-        removals = [
-            removal for removal in spec_diff.get("iterable_item_removed", []) if removal.up.path(output_format="list")[-1] == "enum"
-        ]
-        assert len(removals) == 0, f"The current spec narrowed a field enum: {spec_diff.pretty()}"
+    @pytest.fixture(name="spec_diff")
+    def spec_diff_fixture(
+        self,
+        skip_backward_compatibility_tests: bool,
+        actual_connector_spec: ConnectorSpecification,
+        previous_connector_spec: ConnectorSpecification,
+    ) -> DeepDiff:
+        assert isinstance(actual_connector_spec, ConnectorSpecification) and isinstance(previous_connector_spec, ConnectorSpecification)
+        return
 
     @pytest.mark.default_timeout(60)
     @pytest.mark.backward_compatibility
-    def test_new_enum_field_declaration(self, spec_diff: DeepDiff):
-        """Check if an 'enum' field was added to the spec."""
-        added_enum_fields = [
-            addition for addition in spec_diff.get("dictionary_item_added", []) if addition.path(output_format="list")[-1] == "enum"
-        ]
-        assert len(added_enum_fields) == 0, f"An 'enum' field was declared on an existing property of the spec: {spec_diff.pretty()}"
+    def test_backward_compatibility(
+        self,
+        skip_backward_compatibility_tests: bool,
+        inputs,
+        actual_connector_spec: ConnectorSpecification,
+        previous_connector_spec: ConnectorSpecification,
+    ):
+        """Check if the current spec is backward_compatible:
+        1. Perform multiple hardcoded syntactic checks with SpecDiffChecker.
+        2. Validate fake generated previous configs against the actual connector specification with validate_previous_configs.
+        """
+        assert isinstance(actual_connector_spec, ConnectorSpecification) and isinstance(previous_connector_spec, ConnectorSpecification)
+        spec_diff = self.compute_spec_diff(actual_connector_spec, previous_connector_spec)
```

Comment:
I decided to discard `spec_diff` fixture for the moment because:
* No other tests are using it.
* The signature of `test_backward_compatibility` was changed to use `actual_connector_spec` and `previous_connector_spec` (required for `validate_previous_configs`). Using a `spec_diff` fixture was convenient to have a lighter signature on `test_backward_compatibility` that only required `spec_diff` fixture, but now that I added `validate_previous_configs`, I find it overkill to add a dedicated `spec_diff` fixture that takes the same inputs as the actual test.